packages/terser/src/index.js

```diff
@@ -1,8 +1,76 @@
+ * TODO: maybe we should generalize this to a package which would be useful outside
+ *       bazel; however we would have to support the full terser CLI and not make
+ *       assumptions about how the argv looks.
+ */
+const fs = require('fs');
+const path = require('path');
+const child_process = require('child_process');
 
-// Pass-through require, ensures that the nodejs_binary will load the version of terser
-// from @bazel/terser package.json, not some other version the user depends on.
-require('terser/bin/uglifyjs');
+// Run Bazel with --define=VERBOSE_LOGS=1 to enable this logging
+const VERBOSE_LOGS = !!process.env['VERBOSE_LOGS'];
 
-// TODO: add support for minifying multiple files (eg. a TreeArtifact) in a single execution
-// Under Node 12 it should use the worker threads API to saturate all local cores
\ No newline at end of file
+function log_verbose(...m) {
+  if (VERBOSE_LOGS) console.error('[terser/index.js]', ...m);
+}
+
+// Peek at the arguments to find any directories declared as inputs
+let argv = process.argv.slice(2);
+// terser_minified.bzl always passes the inputs first,
+// then --output [out], then remaining args
+// We want to keep those remaining ones to pass to terser
+// Avoid a dependency on a library like minimist; keep it simple.
+const outputArgIndex = argv.findIndex((arg) => arg.startsWith('--'));
+
+const inputs = argv.slice(0, outputArgIndex);
+const output = argv[outputArgIndex + 1];
+const residual = argv.slice(outputArgIndex + 2);
+
+log_verbose(`Running terser/index.js
+  inputs: ${inputs}
+  output: ${output}
+  residual: ${residual}`);
+
+function isDirectory(input) {
+  return fs.lstatSync(path.join(process.cwd(), input)).isDirectory();
+}
+
+function terserDirectory(input) {
+  if (!fs.existsSync(output)) {
+    fs.mkdirSync(output);
+  }
+
+  fs.readdirSync(input).forEach(f => {
+    if (f.endsWith('.js')) {
+      const inputFile = path.join(input, path.basename(f));
+      const outputFile = path.join(output, path.basename(f));
+      // We don't want to implement a command-line parser for terser
+      // so we invoke its CLI as child processes, just altering the input/output
+      // arguments. See discussion: https://github.com/bazelbuild/rules_nodejs/issues/822
+      // FIXME: this causes unlimited concurrency, which will definitely eat all the RAM on your
+      // machine;
+      //        we need to limit the concurrency with something like the p-limit package.
+      // TODO: under Node 12 it should use the worker threads API to saturate all local cores
+      child_process.fork(
+          // __dirname will be the path to npm_bazel_terser (Bazel's name for our src/ directory)
+          // and our node_modules are installed in the ../npm directory since they're external
+          // Note that the fork API doesn't do any module resolution.
+          path.join(__dirname, '../npm/node_modules/terser/bin/uglifyjs'),
```

Comment:
couldn't you do something like: `const path = path.join(require.resolve('terser'), 'bin/uglifyjs')`?

Reply:
Actually it was busted when tested from e2e anyway.

I think the simplest thing is since fork doesn't do module resolve, it should be `require.resolve(the_same_thing_as_below)`

Fixed, thx